### PR TITLE
Change to building the image outside of compose

### PR DESCRIPTION
**Pull request by NCarlsonMSFT**

### CODE REVIEW

File: generators/app/templates/_dockerTaskGeneric.sh [hunk 1]
```diff
@@ -4,7 +4,7 @@ isWebProject=<%= isWebProject %>
 
 # Kills all running containers of an image and then removes them.
 cleanAll () {
-    # List all running containers that use $imageName, kill them and then remove them.
+    # List all running containers that use $serviceName, kill them and then remove them.
```
Comment (PeterJausovec) [inline generators/app/templates/_dockerTaskGeneric.sh]: You are not using $serviceName in cleanAll.. typo?

Comment (NCarlsonMSFT) [inline generators/app/templates/_dockerTaskGeneric.sh]: That was a mistake, with the opposite mistake in the .sh file :/

---
In reply to: [66712237](https://github.com/Microsoft/generator-docker/pull/63#discussion_r66712237) [](ancestors = 66712237)

File: generators/app/templates/docker-compose.yml [hunk 2]
```diff
@@ -1,9 +1,12 @@
...
-    - "<%= portNumber %>:<%= portNumber %>"<% } %><% if (environment === 'debug' && volumeMap !== null) { %>
-  volumes:
-    - <%= volumeMap %><% } %>
-  labels:
-    com.<%= imageName %>.environment: "<%= environment %>"
+version: '2'
+
+services:
+  <%= serviceName %>:<% if (environment === 'debug') { %>
+    image: <%= imageName %>:debug<% } else { %>
```
Comment (saurabsa) [inline generators/app/templates/docker-compose.yml]: Does it need to be in capital "Debug" or is it case insensitive?

Comment (NCarlsonMSFT) [inline generators/app/templates/docker-compose.yml]: It is case sensitive (That's why i added a ToLower in the PS1) and the files are also lower case and case sensitive in the Mac scenario so i am trying to stay lowercase consistently. 

---
In reply to: [66846684](https://github.com/Microsoft/generator-docker/pull/63#discussion_r66846684) [](ancestors = 66846684)

**Comment by saurabsa** [comment]
:shipit:

**Pull request merged**